src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -33368,635 +33971,3631 @@ .maxstack  3
+                    public void Add(int i) { _list.Add(i is T t ? t : default); }
+                }
+                """;
+
+            string sourceB1 = """
+                class Program
+                {
+                    static void Main()
+                    {
+                        int x = 1;
+                        int[] y = [2, 3];
+                        MyCollection<object> z = [x, ..y];
+                        z.Report();
+                    }
+                }
+                """;
+            CompileAndVerify([sourceA, sourceB1, s_collectionExtensions], expectedOutput: "[1, 2, 3], ");
+
+            string sourceB2 = """
+                class Program
+                {
+                    static void Main()
+                    {
+                        string x = "1";
+                        string[] y = ["2", "3"];
+                        MyCollection<object> z = [x, ..y];
+                        z.Report();
+                    }
+                }
+                """;
+            CompileAndVerify([sourceA, sourceB2, s_collectionExtensions], expectedOutput: "[1, 2, 3], ");
+
+            string sourceB3 = """
+                class Program
+                {
+                    static void Main()
+                    {
+                        int? x = 1;
+                        int?[] y = [2, 3];
+                        MyCollection<int?> z = [x, ..y];
+                    }
+                }
+                """;
+            var comp = CreateCompilation([sourceA, sourceB3]);
+            comp.VerifyEmitDiagnostics(
+                // (7,33): error CS1950: The best overloaded Add method 'MyCollection<int?>.Add(string)' for the collection initializer has some invalid arguments
+                //         MyCollection<int?> z = [x, ..y];
+                Diagnostic(ErrorCode.ERR_BadArgTypesForCollectionAdd, "x").WithArguments("MyCollection<int?>.Add(string)").WithLocation(7, 33),
+                // (7,33): error CS1503: Argument 1: cannot convert from 'int?' to 'string'
+                //         MyCollection<int?> z = [x, ..y];
+                Diagnostic(ErrorCode.ERR_BadArgType, "x").WithArguments("1", "int?", "string").WithLocation(7, 33),
+                // (7,36): error CS1503: Argument 1: cannot convert from 'int?' to 'string'
+                //         MyCollection<int?> z = [x, ..y];
+                Diagnostic(ErrorCode.ERR_BadArgType, "..y").WithArguments("1", "int?", "string").WithLocation(7, 36),
+                // (7,38): error CS1950: The best overloaded Add method 'MyCollection<int?>.Add(string)' for the collection initializer has some invalid arguments
+                //         MyCollection<int?> z = [x, ..y];
+                Diagnostic(ErrorCode.ERR_BadArgTypesForCollectionAdd, "y").WithArguments("MyCollection<int?>.Add(string)").WithLocation(7, 38));
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData("in")]
+        [InlineData("ref readonly")]
```

Comment:
Why does this work for `ref readonly`? I would, at the very least, expect a warning?

Reply:
Same response, I guess.